File: tests/machine/x/clj/match_full.clj
(ns main)

(declare x label day mood ok status)

;; Function classify takes [n: int] and returns string
(defn classify [n]
  (try
    (throw (ex-info "return" {:value (let [t n]
  (cond
    (= t 0) "zero"
    (= t 1) "one"
    :else "many"
  ))}))
  (catch clojure.lang.ExceptionInfo e
    (if (= (.getMessage e) "return")
      (:value (ex-data e))
    (throw e)))
  )
)

(defn -main []
  (def x 2) ;; int
  (def label (let [t x]
  (cond
    (= t 1) "one"
    (= t 2) "two"
    (= t 3) "three"
    :else "unknown"
  ))) ;; string
  (println label)
  (def day "sun") ;; string
  (def mood (let [t day]
  (cond
    (= t "mon") "tired"
    (= t "fri") "excited"
    (= t "sun") "relaxed"
    :else "normal"
  ))) ;; string
  (println mood)
  (def ok true) ;; bool
  (def status (let [t ok]
  (cond
    (= t true) "confirmed"
    (= t false) "denied"
    :else nil
  ))) ;; string
  (println status)
  (println (classify 0))
  (println (classify 5))
)

(-main)

File: archived/tests/transpiler/x/fs/match_full.fs
// Generated 2025-07-21 18:37 +0700

let x: int = 2
let label: string = match x with
| 1 -> "one"
| 2 -> "two"
| 3 -> "three"
| _ -> "unknown"
printfn "%s" (string label)
let day: string = "sun"
let mood: string = match day with
| "mon" -> "tired"
| "fri" -> "excited"
| "sun" -> "relaxed"
| _ -> "normal"
printfn "%s" (string mood)
let ok: bool = true
let status: string = match ok with
| true -> "confirmed"
| false -> "denied"
printfn "%s" (string status)
let rec classify n =
    match n with
| 0 -> "zero"
| 1 -> "one"
| _ -> "many"
printfn "%s" (string (classify 0))
printfn "%s" (string (classify 5))
